docs/howto/redis_cluster.md

```diff
@@ -0,0 +1,41 @@
+# Redis Cluster
+
+commcare-cloud supports setting up Redis as a monolith or as a cluster. Setting up monolith is simple and is in built in the full stack deploy. Simply add the host to `redis` host group in the inventory. Setting up a cluster is not fully automated in commcare-cloud and there are few additional steps required. This document explains how to setup Redis cluster.
+
+## How to setup a Redis cluster
+
+Redis requires at least three master nodes for a functional cluster. Read more on Redis cluster tutorial [documentation]([https://redis.io/topics/cluster-tutorial](https://redis.io/topics/cluster-tutorial)) to understand other requirements and limitations of Redis cluster. Redis cluster supports Redis master slave model, but commcare-cloud supports running Redis cluster with master nodes only. To setup Redis cluster using commcare-cloud, follow below steps
+
+- Add Redis hosts under inventory groups `redis` and `redis_cluster_master`
+- Run below to install Redis on the three nodes with cluster specific config
+  ```
+  commcare-cloud <env> deploy-stack --limit=<redis_hosts> --tags=redis
+  ```
+- On any one of the nodes, run below command to make the Redis hosts form a cluster with equal number of hash slots on each node (aka shards)
+   ```
+   /usr/local/src/redis-4.0.8/src/redis-trib.rb create --replicas 0 <list of redis URLs of all hosts in the format redis_ip:redis_port>
```

Comment:
side note: do you think it would be worthwhile for us to add replicas?

Reply:
Replicas make sense for high availability which we don't need for Redis given our use case, I think.